packages/synapse-react-client/package.json

```diff
@@ -111,7 +112,8 @@
     "ua-parser-js": "^1.0.35",
     "universal-cookie": "^4.0.4",
     "use-deep-compare-effect": "^1.8.1",
-    "xss": "^1.0.14"
+    "xss": "^1.0.14",
+    "zustand": "^4.4.1"
```

Comment:
[zustand](https://github.com/pmndrs/zustand) is a state management library that provides far superior performance to React Context for write-many workloads.

There are many options in this space, I chose zustand because it's small and already included in our bundle (used by reactflow, which powers the ProvenanceGraph)